plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerLabelProvider.java

```diff
@@ -63,58 +68,108 @@ public Image getImage(Object element) {
-				return style(resource.getName(), build.getStatus());
+				return getStyledText((IBuild) resource);
 			case ResourceKind.BUILD_CONFIG:
-				return style(resource.getName(), ((IBuildConfig) resource).getSourceURI());
+				return getStyledText((IBuildConfig) resource);
 			case ResourceKind.DEPLOYMENT_CONFIG:
-				IDeploymentConfig config = (IDeploymentConfig) resource;
-				return style(config.getName(), String.format("selector: %s", StringUtils.serialize(config.getReplicaSelector())));
+				return getStyledText((IDeploymentConfig) resource);
 			case ResourceKind.IMAGE_STREAM:
-				IImageStream repo = (IImageStream) resource;
-				return style(repo.getName(), repo.getDockerImageRepository().toString());
+				return getStyledText((IImageStream) resource);
 			case ResourceKind.POD:
-				IPod pod = (IPod) resource;
-				String labels = StringUtils.serialize(pod.getLabels());
-				if(StringUtils.isEmpty(labels)){
-					return new StyledString(pod.getName());
-				}
-				String podQualifiedText = String.format("labels: %s", labels);
-				return style(pod.getName(), podQualifiedText);
+				return getStyledText((IPod) resource);
 			case ResourceKind.PROJECT:
-				IProject project = (IProject) resource;
-				String name = org.apache.commons.lang.StringUtils.defaultIfBlank(project.getDisplayName(), project.getName());
-				return style(name, "");
+				return getStyledText((IProject) element);
 			case ResourceKind.ROUTE:
-				IRoute route = (IRoute) resource;
-				return style(route.getName(), String.format("%s%s", route.getHost(),route.getPath()));
+				return getStyledText((IRoute) resource);
 			case ResourceKind.REPLICATION_CONTROLLER:
-				IReplicationController rc = (IReplicationController) resource;
-				return (style(resource.getName(), String.format("selector: %s", StringUtils.serialize(rc.getReplicaSelector()))));
+				return getStyledText((IReplicationController) resource);
 			case ResourceKind.SERVICE:
 				IService service = (IService) resource;
-				String serviceQualifiedText = String.format("selector: %s", StringUtils.serialize(service.getSelector()));
-				return style(service.getName(), serviceQualifiedText);
+				return getStyledText(service);
+			case ResourceKind.TEMPLATE:
+				return getStyledText((ITemplate) element);
 			default:
 				break;
 			}
 		}
 		if (element instanceof ResourceGrouping) {
-			return new StyledString(StringUtils.humanize(((ResourceGrouping) element).getKind() + "s"));
+			return getStyledText((ResourceGrouping) element);
 		} else if (element instanceof Connection) {
-			Connection conn = (Connection) element;
-			String prefix = org.apache.commons.lang.StringUtils.defaultIfBlank(conn.getUsername(), "<unknown user>");
-			if(prefix == null) {
-				prefix = "<unknown user>";
-			}
-			return style(prefix, conn.toString());
+			return getStyledText((Connection) element);
 		}
 		return super.getStyledText(element);
 	}
 
+	private StyledString getStyledText(IService service) {
```

Comment:
Should we move these to some utility method to allow for testing and reduce the size of the class?